### PR TITLE
add Unified Object Model Proposal

### PR DESCRIPTION
**Pull request by lguohan**

### CODE REVIEW

File: doc/SAI-Proposal-15-UOM.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+       _In_ const sai_attribute_t *attr
+       );
+ 
+    typedef sai_status_t (*sai_get_next_hop_group_member_attribute_fn)(
+       _In_ sai_object_id_t next_hop_group_member_id,
+       _In_ const uint32_t attr_count,
+       _Inout_ sai_attribute_t *attr_list
+       );
+
+In the nexthopgroup object, remove SAI_NEXT_HOP_GROUP_ATTR_NEXT_HOP_LIST attribute.
```
Comment (atitjain) [inline doc/SAI-Proposal-15-UOM.md]: please retain that as a read only attribute.

File: doc/SAI-Proposal-15-UOM.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+       );
+ 
+    typedef sai_status_t (*sai_get_vlan_member_attribute_fn)(
+       _In_ sai_object_id_t vlan_member_id,
+       _In_ const uint32_t attr_count,
+       _Inout_ sai_attribute_t *attr_list
+       );
+
+In the VLAN object, add SAI_VLAN_ATTR_MEMBER_LIST to query the vlan members. remove SAI_VLAN_ATTR_PORT_LIST. 
+
```
Comment (atitjain) [inline doc/SAI-Proposal-15-UOM.md]: Consider retaining it as a read only attribute. Will be helpful for the host adapter to get the list if it needs.

File: doc/SAI-Proposal-15-UOM.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+This set of SAI objects (later referred as SAI view) can be used for many purposes. For example, we can use these SAI objects to capture current ASIC states and later used for restore the ASIC states after ASIC re-initialization. We can also compare two sets of SAI objects and find their differences and then make transition from one ASIC state for another ASIC state. 
+
+Most of SAI objects are now using Unified Object ID and CRUD API to manage them. However, some SAI objects are still having specific APIs to manage. For example, VLAN has add/remove ports API to add/remove ports from VLAN. These APIs complicates the operations needed to restore ASIC states from a SAI view, or transition from one SAI view to another SAI view, because the program needs to use these specific APIs to restore or make a transition. This also applies to port break out operation where specific attributes is used for such operation.
```
Comment (atitjain) [inline doc/SAI-Proposal-15-UOM.md]: What about the objects such as FDB, route and Neighbour which still dont have object_id associated and have to be handled thorough the key. In the warmboot proposal we have different getter/bulk getter APIs for each of those.

File: doc/SAI-Proposal-15-UOM.md (hunk 1 above)
Comment (lguohan) [inline doc/SAI-Proposal-15-UOM.md]: add new RO SAI_NEXT_HOP_GROUP_ATTR_NEXT_HOP_MEMBER_LIST attribute in pull request https://github.com/opencomputeproject/SAI/pull/145

File: doc/SAI-Proposal-15-UOM.md (hunk 2 above)
Comment (lguohan) [inline doc/SAI-Proposal-15-UOM.md]: added in new pull request https://github.com/opencomputeproject/SAI/pull/146

File: doc/SAI-Proposal-15-UOM.md (hunk 3 above)
Comment (lguohan) [inline doc/SAI-Proposal-15-UOM.md]: those objects are probably not going to be referenced by any other sai objects, thus having object_id for them is not a MUST.

File: doc/SAI-Proposal-15-UOM.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+### PORT
+
+Use the port creation/removal API to do port breakout/break-in operation. SAI_PORT_ATTR_HW_LANE_LIST and SAI_PORT_ATTR_SPEED are mandatory attributes for port creation. Breakout can be achieved using following sequence, 1) remove a port, 2) create port for each lane.
+
+    typedef enum _sai_port_attr_t
+    {
+        /** READ-WRITE */
+       
+		/** Hardware Lane list [sai_u32_list_t]
```
Comment (stcheng) [inline doc/SAI-Proposal-15-UOM.md]: extra space

**Comment by lguohan** [comment]
will later rewrite the document

**Pull request merged**